boards: arm: frdm_kw41z: Fix pwm/tpm pin settings

galak opened the pull request:
The nodelabel for the PWM controller on KW41Z is TPM[0..n], so we need
to change the logic in the ifdef to get the pins setup.

Additionally the setup of PORTA0/1 and PORTB16/17 pins for TPM only
exists on kPORT_MuxAlt5.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
boards: arm: frdm_kw41z: Fix pwm/tpm pin settings

The nodelabel for the PWM controller on KW41Z is TPM[0..n], so we need
to change the logic in the ifdef to get the pins setup.

Additionally the setup of PORTA0/1 and PORTB16/17 pins for TPM only
exists on kPORT_MuxAlt5.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak commented:
@alexandru-porosanu-nxp what was trying to be done with PTA0/1 and PTB16/17?

alexandru-porosanu-nxp commented:
> @alexandru-porosanu-nxp what was trying to be done with PTA0/1 and PTB16/17?

I don't think I've ever tested the TPM on  PTA and PTB, so I guess the mux setting off-by-one in my settings. Unfortunately, I don't have a KW41 board handy to test this, but as long as PTC mux setting is the same, this should work as before.

galak commented:
> I don't think I've ever tested the TPM on PTA and PTB, so I guess the mux setting off-by-one in my settings. Unfortunately, I don't have a KW41 board handy to test this, but as long as PTC mux setting is the same, this should work as before.

As this was just for testing purposes any issue with me removing the pin settings for PTA0/1 and PTB16/17 since there isn't any code that is utilizing them in Zephyr?

alexandru-porosanu-nxp commented:
> > I don't think I've ever tested the TPM on PTA and PTB, so I guess the mux setting off-by-one in my settings. Unfortunately, I don't have a KW41 board handy to test this, but as long as PTC mux setting is the same, this should work as before.
> 
> As this was just for testing purposes any issue with me removing the pin settings for PTA0/1 and PTB16/17 since there isn't any code that is utilizing them in Zephyr?

Yep, that's fine with me.

galak committed:
boards: arm: frdm_kw41z: Remove testing PWM pins

The TPM1/2 pin settings aren't used by any drivers or other devices and
only existed for testing purposes.  Remove them since nothing in tree
is utilizing these settings.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak commented:
> > > I don't think I've ever tested the TPM on PTA and PTB, so I guess the mux setting off-by-one in my settings. Unfortunately, I don't have a KW41 board handy to test this, but as long as PTC mux setting is the same, this should work as before.
> > 
> > 
> > As this was just for testing purposes any issue with me removing the pin settings for PTA0/1 and PTB16/17 since there isn't any code that is utilizing them in Zephyr?
> 
> Yep, that's fine with me.

Cool.  I added a commit to remove the settings.
